lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -370,7 +370,7 @@ private void readAffixFile(InputStream affixStream, CharsetDecoder decoder)
         fullStrip = true;
       } else if ("LANG".equals(firstWord)) {
         language = singleArgument(reader, line);
-        alternateCasing = "tr_TR".equals(language) || "az_AZ".equals(language);
```

Comment:
Hmm... I didn't check but can we use Locale to parse it properly to extract the language from country variant?

Reply:
Probably we can, but `Locale`s javadoc is so scary and the class looks so general that I'd prefer this simple code.

Reply:
Locale reflects what those iso codes are in the wild... That's why I'm a bit uneasy about using a simple prefix... It should at least be something like this: language.equalsIgnoreCase("tr") || language.toLowerCase(Locale.ROOT).startsWith("tr_")

Reply:
Good point. So it seems that some parsing is inevitable, and quick googling shows that JDK doesn't seem to have that parsing. So I'll split by '_' myself.

Reply:
I looked at that too - surprising there isn't iso -> locale conversion built-in. Sorry.